aws-api/src/main/java/com/amplifyframework/api/aws/GsonVariablesSerializer.java

```diff
@@ -46,11 +46,28 @@ public String serialize(Map<String, Object> variables) {
         return gson.toJson(variables);
     }
 
-    /**
-     * Serializer of {@link Temporal.Date}, an extended ISO-8601 Date string, with an optional timezone offset.
-     *
-     * https://docs.aws.amazon.com/appsync/latest/devguide/scalars.html
-     */
+    @Override
+    public boolean equals(Object thatObject) {
+        if (this == thatObject) {
+            return true;
+        }
+        if (thatObject == null || getClass() != thatObject.getClass()) {
```

Comment:
Hmm, could this be simplified to eliminate the fallback `true`? Maybe something like:

```java
if (thatObject == null) {
  return false
}

return this == thatObject || getClass() == thatObject.getClass()
```

Reply:
Logically, yes.

But, I think it's better to keep as is because:
1. It matches the rest of the codebase
2. It's what the IDE generates if you ask it to add equals/hashcode.